internal/pipe/chocolatey/chocolatey.go

```diff
@@ -0,0 +1,340 @@
+
+func (Pipe) String() string                 { return "chocolatey packages" }
+func (Pipe) Skip(ctx *context.Context) bool { return len(ctx.Config.Chocolateys) == 0 }
+
+// Default sets the pipe defaults.
+func (Pipe) Default(ctx *context.Context) error {
+	for i := range ctx.Config.Chocolateys {
+		choco := &ctx.Config.Chocolateys[i]
+
+		if choco.Name == "" {
+			choco.Name = ctx.Config.ProjectName
+		}
+
+		if choco.Title == "" {
+			choco.Title = ctx.Config.ProjectName
+		}
+
+		if choco.Goamd64 == "" {
+			choco.Goamd64 = "v1"
+		}
+
+		if choco.SourceRepo == "" {
+			choco.SourceRepo = "https://push.chocolatey.org/"
+		}
+	}
+
+	return nil
+}
+
+// Run the pipe.
+func (Pipe) Run(ctx *context.Context) error {
+	client, err := client.New(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, choco := range ctx.Config.Chocolateys {
+		if err := doRun(ctx, client, choco); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Publish packages.
+func (Pipe) Publish(ctx *context.Context) error {
+	if ctx.SkipPublish {
+		return pipe.ErrSkipPublishEnabled
+	}
+
+	artifacts := ctx.Artifacts.Filter(
+		artifact.ByType(artifact.PublishableChocolatey),
+	).List()
+
+	for _, artifact := range artifacts {
+		if err := doPush(ctx, artifact); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func doRun(ctx *context.Context, cl client.Client, choco config.Chocolatey) error {
+	filters := []artifact.Filter{
+		artifact.ByGoos("windows"),
+		artifact.ByType(artifact.UploadableArchive),
+		artifact.Or(
+			artifact.And(
+				artifact.ByGoarch("amd64"),
+				artifact.ByGoamd64(choco.Goamd64),
+			),
+			artifact.ByGoarch("386"),
+		),
+	}
+
+	if len(choco.IDs) > 0 {
+		filters = append(filters, artifact.ByIDs(choco.IDs...))
+	}
+
+	artifacts := ctx.Artifacts.
+		Filter(artifact.And(filters...)).
+		List()
+
+	if len(artifacts) == 0 {
+		return errors.New("chocolatey requires a windows build and archive")
+	}
+
+	// folderDir is the directory that then will be compressed to make the
+	// chocolatey package.
+	folderPath := filepath.Join(ctx.Config.Dist, choco.Name+".choco")
+	toolsPath := filepath.Join(folderPath, "tools")
+	if err := os.MkdirAll(toolsPath, 0o755); err != nil {
+		return err
+	}
+
+	nuspecFile := filepath.Join(folderPath, choco.Name+".nuspec")
+	nuspec, err := buildNuspec(ctx, choco)
+	if err != nil {
+		return err
+	}
+
+	if err = os.WriteFile(nuspecFile, nuspec, 0o644); err != nil {
+		return err
+	}
+
+	data, err := dataFor(ctx, cl, choco, artifacts)
+	if err != nil {
+		return err
+	}
+
+	script, err := buildTemplate(choco.Name, scriptTemplate, data)
+	if err != nil {
+		return err
+	}
+
+	scriptFile := filepath.Join(toolsPath, "chocolateyinstall.ps1")
+	log.WithField("file", scriptFile).Debug("creating")
+	if err = os.WriteFile(scriptFile, script, 0o644); err != nil {
+		return err
+	}
+
+	log.WithField("nuspec", nuspecFile).Info("packing")
+	out, err := cmd.Exec(ctx, "choco.exe", "pack", nuspecFile, "--out", ctx.Config.Dist)
```

Comment:
this will only ever work on windows, right?

maybe we could also add a `runtime.GOOS != "windows"` to `Skip`? What do you think?

Reply:
Not exactly, we could install `chocolatey` in Linux also. Here an example:
https://github.com/faabiosr/rand-beer/blob/main/.github/workflows/release.yml#L17

Reply:
and the binary is still called choco.exe... well... lol haha

Reply:
Let me confirm running in both OS's.

Reply:
Well, it is possible to remove the `.exe` extension. What do you think, remove or not? 😄

Reply:
I think if it works on both windows and unix without the `.exe`, I would probably remove it yes

Reply:
Okay, I will remove then.

Reply:
Done!